src/main/java/io/aiven/kafka/connect/opensearch/BulkProcessor.java

```diff
@@ -401,6 +421,25 @@ private BulkResponse execute() throws Exception {
             }, maxRetries, retryBackoffMs, RuntimeException.class);
         }
 
+        private void reportMalformedRecord(final BulkItemResponse itemResponse) {
+            try {
+                if (reporter != null) {
+                    LOGGER.info("Reporting on record failure to DLQ");
+                    final SinkRecord record = batch.get(itemResponse.getItemId()).getSinkRecord();
+                    if (!(record.value() instanceof Map)) {
+                        LOGGER.error("Only Map objects supported");
```

Comment:
Why do we have this check here? Doesn't the connector support arrays and structs as well?

Reply:
This was here by mistake, removed it.